support-frontend/assets/helpers/subscriptionsForms/formReducer.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+// ----- Reducer ----- //
+
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import { getUser } from 'pages/digital-subscription-checkout/helpers/user';
+import type { BillingPeriod } from 'helpers/billingPeriods';
+import { defaultPaymentMethod } from 'helpers/subscriptionsForms/countryPaymentMethods';
+import type { SubscriptionProduct } from 'helpers/subscriptions';
+import { isTestUser } from 'helpers/user/user';
+import type { Action } from 'helpers/subscriptionsForms/formActions';
+import { removeError } from 'helpers/subscriptionsForms/validation';
+import type { ProductOptions } from 'helpers/productPrice/productOptions';
+import { NoProductOptions } from 'helpers/productPrice/productOptions';
+import type { FulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
+import { NoFulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
+import type { FormState } from 'helpers/subscriptionsForms/formFields';
+import type { Option } from 'helpers/types/option';
+
+function createFormReducer(
+  initialCountry: IsoCountry,
+  product: SubscriptionProduct,
+  initialBillingPeriod: BillingPeriod,
+  startDate: Option<string>,
+  productOption: Option<ProductOptions>,
+  fulfilmentOption: Option<FulfilmentOptions>,
+) {
+  const user = getUser(); // TODO: Is this unnecessary? It could use the user reducer
```

Comment:
I just looked at the initial state of the page and these details don't seem to be in the user object, but they are in the checkout object.

Reply:
Yeah I think it isn't but I'm going to leave changing it for a future PR

Reply:
👍